typedb/stream/bidirectional_stream.py

```diff
@@ -127,7 +131,9 @@ def __init__(self, request_id: UUID, stream: "BidirectionalStream"):
             self._stream = stream
 
         def get(self) -> T:
-            return self._stream.fetch(self._request_id)
+            value = self._stream.fetch(self._request_id)
+            self._stream.done(self._request_id)
```

Comment:
a `Single` can immediately remove itself from the transaction stream when the user retrieves the value via `get()`